catkin_tools/argument_parsing.py

```diff
@@ -201,32 +241,75 @@ def handle_make_arguments(input_make_args, force_single_threaded_when_running_te
     """
     make_args = list(input_make_args)
 
+    # Get the values for the jobs flags which may be in the make args
+    jobs_dict = extract_jobs_flags_values(' '.join(make_args))
+
     if force_single_threaded_when_running_tests:
         # force single threaded execution when running test since rostest does not support multiple parallel runs
         run_tests = [a for a in make_args if a.startswith('run_tests')]
         if run_tests:
             wide_log('Forcing "-j1" for running unit tests.')
-            make_args.append('-j1')
-
-    # If no -j/--jobs/-l/--load-average flags are in make_args
-    if not extract_jobs_flags(' '.join(make_args)):
-        # If -j/--jobs/-l/--load-average are in MAKEFLAGS
-        if 'MAKEFLAGS' in os.environ and extract_jobs_flags(os.environ['MAKEFLAGS']):
-            # Do not extend make arguments, let MAKEFLAGS set things
-            pass
-        else:
-            # Else extend the make_arguments to include some jobs flags
-            # Use the number of CPU cores
-            try:
-                jobs = cpu_count()
-                make_args.append('-j{0}'.format(jobs))
-                make_args.append('-l{0}'.format(jobs))
-            except NotImplementedError:
-                # If the number of cores cannot be determined, do not extend args
-                pass
+            jobs_dict['jobs'] = 1
+
+    if 'jobs' in jobs_dict:
+        make_args.append('-j{0}'.format(jobs_dict['jobs']))
+    if 'load-average' in jobs_dict:
+        make_args.append('-l{0}'.format(jobs_dict['load-average']))
+
     return make_args
 
 
+def configure_make_args(make_args, internal_make_jobserver):
+    """Initialize the internal GNU Make jobserver or configure it as a pass-through
+
+    :param make_args: arguments to be passed to GNU Make
+    :type make_args: list
+    :param internal_make_jobserver: if true, use the internal jobserver
+    :type make_args: bool
+    :rtype: tuple
+    """
```

Comment:
The return tuple of this function is complicated, I think it would be really helpful for others if what is in it is spelled out in the `:returns:` directive or in the body of the doc string.